### ISSUE TITLE
False negatives for length-zero-no-unit

### ISSUE THREAD

**Issue by ncroll** [stylelint/stylelint#5558]
> Clearly describe the bug

The [length-zero-no-unit](https://stylelint.io/user-guide/rules/list/length-zero-no-unit/) lint gives false negatives in a couple of scenarios related to flexbox.

> Which rule, if any, is the bug related to?

`length-zero-no-unit`

> What code is needed to reproduce the bug?

This fails even though the units are needed to disambiguate that we're using the shorthand for `flex-basis`:
```css
.foo {
  flex: 0px; 
}
```

This fails even though units are required on `flex-basis` for IE 11 support:
```css
.bar {
  flex: 1 1 0px;
}
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "rules": {
    "length-zero-no-unit": true
  }
}
```

> Which version of stylelint are you using?

e.g. `13.11.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

Node.js API

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

e.g. "The following warnings were flagged:"

```shell
test.scss
test.scss: line 7, col 14 - Unexpected unit (length-zero-no-unit)
```

**Comment by ybiquitous** [comment]
@ncroll Thanks for reporting the issue. This has been resolved via #5315. Can you try the [v14](https://github.com/stylelint/stylelint/tree/v14) branch, please?
